server/main.lua

```diff
@@ -0,0 +1,101 @@
+---@class vehicleData
+---@field model string
+---@field plate string
+---@field mods string
+
+---@class playerData
+---@field license string
+---@field citizenId string
+
+
+--- Creates a Vehicle DB Entity
+---@param playerData playerData
+---@param vehicleData vehicleData
+local function createEntity(playerData, vehicleData)
+    MySQL.insert('INSERT INTO player_vehicles (license, citizenid, vehicle, hash, mods, plate, state) VALUES (?,?,?,?,?,?,?)', {
+        playerData.license,
+        playerData.citizenId,
+        vehicleData.model,
+        joaat(vehicleData.model),
+        vehicleData.mods or '{}',
+        vehicleData.plate,
+        0
+    })
+end
+
+exports('CreateVehicleEntity', createEntity)
+
+---@class FetchVehicleEntityQuery
+---@field valueType 'citizenid'|'license'|'plate'
+---@field value string
+
+---@alias vehicleEntity table
+
+--- Fetches DB Vehicle Entity
+---@param query FetchVehicleEntityQuery
+---@return vehicleData[]|nil
+local function fetchEntity(query)
+    local vehicleData = {}
+    if query.valueType ~= 'citizenid' and query.valueType ~= 'license' and query.valueType ~= 'plate' then return end
+    local results = MySQL.query.await('SELECT * FROM player_vehicles WHERE ? = ?', {
+        query.valueType,
+        query.value
+    })
+    for _, data in ipairs(results) do
+        vehicleData[#vehicleData + 1] = {
+            id = data.id,
+            citizenid = data.citizenid,
+            model = data.vehicle,
+        }
+    end
+    return vehicleData
+end
+
+exports('FetchVehicleEntity', fetchEntity)
+
+---@class UpdateEntityVehicleQuery
+---@field valueType string
+---@field value string
+
+--- Updates a DB Vehicle Entity
+---@param query UpdateEntityVehicleQuery
+---@param vehiclePlate string
+local function updateEntity(query, vehiclePlate)
+    MySQL.update('UPDATE player_vehicles SET ? = ?, WHERE plate = ?', {
+        query.valueType,
+        query.value,
+        vehiclePlate
+    })
+end
+
+exports('UpdateVehicleEntity', updateEntity)
+
+--- Update Vehicle Entity Owner
+---@param citizenId string
+---@param license string
+---@param vehiclePlate string
+local function updateEntityOwner(citizenId, license, vehiclePlate)
```

Comment:
consider making param as a table. I think using a table for API params is a good practice as it will be flexible to future changes.